app/src/consumer/parser_impl_con.c

```diff
@@ -117,10 +117,18 @@ const char *parser_getErrorDescription(parser_error_t err) {
             return "Requiered field serial";
         case parser_invalid_url_format:
             return "Invalid url format (expect to start with https:// and not containing query or fragments)";
+        case parser_invalid_url_length:
+            return "Invalid url length (max 64 characters)";
         case parser_invalid_email_format:
             return "Invalid email format";
+        case parser_invalid_email_length:
+            return "Invalid email length (max 32 characters)";
         case parser_invalid_handle_format:
             return "Invalid handle format";
+        case parser_invalid_handle_length:
+            return "Invalid handle length (max 32 characters)";
+        case parser_invalid_name_length:
+            return "Invalid name length (max 51 characters)";
```

Comment:
The maximum is actually 50 characters:

https://github.com/Zondax/ledger-oasis/blob/6417680c1faf85d1277154b4a3945686abb0cdd6/app/src/consumer/parser_txdef_con.h#L31

```suggestion
            return "Invalid name length (max 50 characters)";
```

Reply:
The maximum is actually 49. If the allocation is 50, we need zero termination to be safe.

Reply:
So, the max for the name field should be 50 characters as this is what is defined by the spec:
https://github.com/oasisprotocol/metadata-registry/blob/production/docs/adr/0001-entity-metadata.md#entity-metadata-statements.

Also, from my testing, names up to 50 characters are currently accepted and a name with 51 characters is properly rejected.

Reply:
This is not the problem.
While CBOR strings might NOT be zero terminated.. when using functions C string functions like sprintf  they ARE zero-terminated. This means that to avoid any issues when operating with these strings.. we MUST allocate AT LEAST one more byte that must be clear up with zeros. This ensure that under no condition, we will have missing NULL termination issues.
If the maximum length in CBOR is 50, then our buffer should be AT LEAST 51 and fully MEMZERO'ed before copying data from CBOR.

@rllola  ^^^^

Reply:
I have played a bit with `snprintf` this week end to see what would be the behavior and I am convinced that we should add the null byte at the end and therefore allocate the extra byte for those.

When I try to `snprintf` a 4 bytes char array of  size 4 (so no null termination) and then I do `printf` I have a non printable char at the end. I guess we don't see it in the ledger because it is able only to print some ASCII char ?

Making sure we terminate with a null byte should not have any impact on the current behavior.